src/connector/src/parser/scalar_adapter.rs

```diff
@@ -157,81 +160,92 @@ impl ScalarAdapter<'_> {
             (ScalarRefImpl::Utf8(s), _, Kind::Enum(_)) => {
                 ScalarAdapter::Enum(EnumString(s.to_owned()))
             }
+            (ScalarRefImpl::List(list), &Type::NUMERIC_ARRAY, _) => {
+                let mut vec = vec![];
+                for scalar in list.iter() {
```

Comment:
Do you find a pattern here? I think we can do it in a recusive way:
```rust
(ScalarRefImpl::List(list), &Type::NUMERIC_ARRAY, _) => {
        let mut vec = Vec::with_capacity(list.len());
        for datum in list.iter() {
            vec.push(
                datum
                    .map(|scalar| ScalarAdapter::from_scalar(scalar, &Type::NUMERIC))
                    .transpose()?,
            );
        }
        ScalarAdapter::List(vec)
    }
```
That's why I proposed to add a `ScalarAdapter::List(Vec<Option<ScalarAdapter<'a>>>)`, the goal is to simplify the implementation.

Reply:
Done